client/lib/signup/site-type.js

```diff
@@ -150,8 +135,40 @@ export function getAllSiteTypes() {
 			siteTopicHeader: i18n.translate( 'What type of products do you sell?' ),
 			siteTopicLabel: i18n.translate( 'What type of products do you sell?' ),
 			customerType: 'business',
-			// TODO: Re-enable "Purchase required" badge, but hide for Jetpack onboarding.
-			// purchaseRequired: true,
+			purchaseRequired: context === CONTEXT_WPCOM ? true : null,
 		},
 	];
 }
+
+/**
+ * Returns site types configuration used for WordPress.com signup
+ *
+ * @return {array} List of WordPress.com site types
+ */
+export const getWpcomSiteTypes = () => getAllSiteTypes( CONTEXT_WPCOM );
+
+/**
+ * Returns site types configuration used for Jetpack onboarding
+ *
+ * @return {array} List of Jetpack site types
+ */
+export const getJetpackSiteTypes = () => getAllSiteTypes( CONTEXT_JETPACK );
+
+/**
+ * Looks up site types array for item match and returns a property value
+ *
+ * @example
+ * // Find the site type where `id === 2`, and return the value of `slug`
+ * const siteTypeValue = getSiteTypePropertyValue( 'id', 2, 'slug' );
+ *
+ * @param {string} key A property name of a site types item
+ * @param {string|number} value The value of `key` with which to filter items
+ * @param {string} property The name of the property whose value you wish to return
+ * @param {array} siteTypes (optional) A site type collection
+ * @return {(string|int)?} value of `property` or `null` if none is found
+ */
+export function getSiteTypePropertyValue( key, value, property, siteTypes = getWpcomSiteTypes() ) {
```

Comment:
In all the instances I could find, `getSiteTypePropertyValue` is used in the context of wpcom onboarding, so this defaults to using wpcom site types.